flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/BufferManager.java

```diff
@@ -129,7 +132,6 @@ private boolean shouldContinueRequest(BufferPool bufferPool) {
 	 */
 	void requestExclusiveBuffers() throws IOException {
 		Collection<MemorySegment> segments = globalPool.requestMemorySegments();
-		checkArgument(!segments.isEmpty(), "The number of exclusive buffers per channel should be larger than 0.");
```

Comment:
this change should be together with the previous hotfix commit " making `initialCredit` as final"

Reply:
Before the last commit, we can't set numExclusive buffers to 0, so the check should be reserved before the last commit.

Reply:
Got it. So do you think we add the `if (initialCredit > 0)` before calling this method inside `RemoteInputChannel#assignExclusiveSegments`?  Just for not necessary to `synchronized` below for empty segments.